From 47e9fdfeb4ddd71e6d36e406e1840c7e1a97dc6d Mon Sep 17 00:00:00 2001
From: Marcos Paulo de Souza <mpdesouza@suse.com>
Date: Fri, 10 May 2024 10:03:43 -0300
Subject: [PATCH] compiler_attributes.h: Disable __compiletime_error macros

There is a bug[1] on LLVM which is used by clang-extract, so this avoid the
problem for now.

[1]: https://github.com/llvm/llvm-project/issues/87588

Signed-off-by: Marcos Paulo de Souza <mpdesouza@suse.com>
---
 include/linux/compiler_attributes.h | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/include/linux/compiler_attributes.h b/include/linux/compiler_attributes.h
index 28566624f008..5b3da103ab64 100644
--- a/include/linux/compiler_attributes.h
+++ b/include/linux/compiler_attributes.h
@@ -148,7 +148,8 @@
  *   gcc: https://gcc.gnu.org/onlinedocs/gcc/Common-Function-Attributes.html#index-error-function-attribute
  */
 #if __has_attribute(__error__)
-# define __compiletime_error(msg)       __attribute__((__error__(msg)))
+//# define __compiletime_error(msg)       __attribute__((__error__(msg)))
+# define __compiletime_error(msg)
 #else
 # define __compiletime_error(msg)
 #endif
-- 
2.44.0

